microstream-storage-embedded-configuration-05.00.02-MS-GA is an invalid module

## Environment Details
* Helidon Version: 3.1.1
* Helidon SE, Helidon MP
* JDK version: N/A
* OS: N/A
* Docker version (if applicable): N/A

----------

## Problem Description
microstream-storage-embedded-configuration is a transitive dependency

```
[INFO] +- io.helidon.integrations.microstream:helidon-integrations-microstream:jar:3.1.1:compile
[INFO] |  +- one.microstream:microstream-storage-embedded:jar:05.00.02-MS-GA:compile
[INFO] |  |  \- one.microstream:microstream-storage:jar:05.00.02-MS-GA:compile
[INFO] |  |     +- one.microstream:microstream-afs-nio:jar:05.00.02-MS-GA:compile
[INFO] |  |     |  \- one.microstream:microstream-afs:jar:05.00.02-MS-GA:compile
[INFO] |  |     \- one.microstream:microstream-persistence-binary:jar:05.00.02-MS-GA:compile
[INFO] |  |        \- one.microstream:microstream-persistence:jar:05.00.02-MS-GA:compile
[INFO] |  +- one.microstream:microstream-storage-embedded-configuration:jar:05.00.02-MS-GA:compile
```

Its module descriptor exports a non-existent package

```
Caused by: java.lang.module.InvalidModuleDescriptorException: Package one.microstream.storage.embedded.configuration not found in module
	at java.base/jdk.internal.module.ModuleInfo.invalidModuleDescriptor(ModuleInfo.java:1212)
	at java.base/jdk.internal.module.ModuleInfo.doRead(ModuleInfo.java:330)
	at java.base/jdk.internal.module.ModuleInfo.read(ModuleInfo.java:129)
	at java.base/jdk.internal.module.ModulePath.readJar(ModulePath.java:652)
	at java.base/jdk.internal.module.ModulePath.readModule(ModulePath.java:331)
```

This problem persists even in 07.01.00-MS-GA. A fix is in place for the next major 08.00.00-MS.

--- a/dependencies/pom.xml
+++ b/dependencies/pom.xml
@@ -122,7 +122,7 @@
         <version.lib.microprofile-rest-client>3.0</version.lib.microprofile-rest-client>
         <version.lib.microprofile-tracing>3.0</version.lib.microprofile-tracing>
         <version.lib.microprofile-lra-api>2.0-RC1</version.lib.microprofile-lra-api>
-        <version.lib.microstream>05.00.02-MS-GA</version.lib.microstream>
+        <version.lib.microstream>08.01.01-MS-GA</version.lib.microstream>
         <version.lib.mongodb.reactivestreams>1.11.0</version.lib.mongodb.reactivestreams>
         <version.lib.mssql-jdbc>8.4.1.jre8</version.lib.mssql-jdbc>
         <version.lib.mysql-connector-j>8.2.0</version.lib.mysql-connector-j>
@@ -734,6 +734,11 @@
             <dependency>
                 <groupId>one.microstream</groupId>
                 <artifactId>microstream-cache</artifactId>
+                <version>${version.lib.microstream}</version>
+            </dependency>
+            <dependency>
+                <groupId>one.microstream</groupId>
+                <artifactId>microstream-persistence-binary-jdk17</artifactId>
                 <version>${version.lib.microstream}</version>
             </dependency>
 

--- a/integrations/microstream/README.md
+++ b/integrations/microstream/README.md
@@ -2,7 +2,7 @@
 
 This projects add [Microstream](https://microstream.one) support to Helidon.
 
-The offical [Microstream documentation](https://manual.docs.microstream.one/) can be found here.
+The official [Microstream documentation](https://manual.docs.microstream.one/) can be found here.
 
 ## helidon-integrations-microstream
 

--- a/integrations/microstream/cache/src/main/java/io/helidon/integrations/microstream/cache/MicrostreamCacheConfigurationBuilder.java
+++ b/integrations/microstream/cache/src/main/java/io/helidon/integrations/microstream/cache/MicrostreamCacheConfigurationBuilder.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021 Oracle and/or its affiliates.
+ * Copyright (c) 2021, 2024 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -16,9 +16,6 @@
 
 package io.helidon.integrations.microstream.cache;
 
-import java.lang.reflect.Field;
-import java.util.function.Predicate;
-
 import javax.cache.configuration.CacheEntryListenerConfiguration;
 import javax.cache.configuration.Factory;
 import javax.cache.expiry.ExpiryPolicy;
@@ -33,6 +30,7 @@ import one.microstream.cache.types.CacheConfigurationBuilderConfigurationBased;
 import one.microstream.cache.types.CacheConfigurationPropertyNames;
 import one.microstream.cache.types.EvictionManager;
 import one.microstream.configuration.types.Configuration;
+import one.microstream.persistence.binary.util.SerializerFoundation;
 
 /**
  * Builder for Microstream-CacheConfigurations.
@@ -89,7 +87,7 @@ public class MicrostreamCacheConfigurationBuilder<K, V>
      *
      * @param <K> type of the cache key
      * @param <V> type of the cache value
-     * @param config helidon configuation
+     * @param config helidon configuration
      * @param keyType type of the cache key
      * @param valueType type of the cache value
      * @return a new CacheConfiguration builder
@@ -171,10 +169,8 @@ public class MicrostreamCacheConfigurationBuilder<K, V>
     }
 
     @Override
-    public MicrostreamCacheConfigurationBuilder<K, V> serializerFieldPredicate(
-            Predicate<? super Field> serializerFieldPredicate) {
-        cacheConfigBuilder.serializerFieldPredicate(serializerFieldPredicate);
-        return this;
+    public Builder<K, V> serializerFoundation(SerializerFoundation<?> serializerFoundation) {
+        return cacheConfigBuilder.serializerFoundation(serializerFoundation);
     }
 
     @Override
@@ -185,7 +181,7 @@ public class MicrostreamCacheConfigurationBuilder<K, V>
 
     private static void verifyType(String typeName, Class<?> actualType) {
         if (!typeName.equals(actualType.getTypeName())) {
-            throw new ConfigException("Microstream cache-config type missmatch, expected value from configuration: " + typeName
+            throw new ConfigException("Microstream cache-config type mismatch, expected value from configuration: " + typeName
                                               + " but got: " + actualType.getTypeName());
         }
     }

--- a/integrations/microstream/cache/src/test/java/io/helidon/integrations/microstream/cache/ConfigurationTest.java
+++ b/integrations/microstream/cache/src/test/java/io/helidon/integrations/microstream/cache/ConfigurationTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021 Oracle and/or its affiliates.
+ * Copyright (c) 2021, 2024 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -16,9 +16,7 @@
 
 package io.helidon.integrations.microstream.cache;
 
-import java.lang.reflect.Field;
 import java.util.Map;
-import java.util.function.Predicate;
 
 import javax.cache.configuration.Factory;
 import javax.cache.expiry.ExpiryPolicy;
@@ -65,9 +63,7 @@ class ConfigurationTest {
                   () -> assertThat("getCacheLoaderFactory", cacheConfig.getCacheLoaderFactory(), nullValue()),
                   () -> assertThat("getCacheWriterFactory", cacheConfig.getCacheWriterFactory(), nullValue()),
                   () -> assertThat("getCacheEntryListenerConfigurations",
-                                   cacheConfig.getCacheEntryListenerConfigurations(), emptyIterable()),
-                  () -> assertThat("getSerializerFieldPredicate", cacheConfig.getSerializerFieldPredicate(),
-                                   is(CacheConfiguration.DefaultSerializerFieldPredicate())));
+                                   cacheConfig.getCacheEntryListenerConfigurations(), emptyIterable()));
     }
 
     /**
@@ -154,14 +150,4 @@ class ConfigurationTest {
         assertThat(cacheConfig.getEvictionManagerFactory(), sameInstance(evictionManagerFactory));
     }
 
-    @Test
-    void serializerFieldPredicate() {
-        @SuppressWarnings("unchecked")
-        Predicate<? super Field> serializerFieldPredicate = Mockito.mock(Predicate.class);
-
-        CacheConfiguration<Integer, String> cacheConfig = MicrostreamCacheConfigurationBuilder
-                .builder(Integer.class, String.class).serializerFieldPredicate(serializerFieldPredicate).build();
-
-        assertThat(cacheConfig.getSerializerFieldPredicate(), sameInstance(serializerFieldPredicate));
-    }
 }

--- a/integrations/microstream/cdi/src/main/java/io/helidon/integrations/microstream/cdi/CacheExtension.java
+++ b/integrations/microstream/cdi/src/main/java/io/helidon/integrations/microstream/cdi/CacheExtension.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021 Oracle and/or its affiliates.
+ * Copyright (c) 2021, 2024 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -100,10 +100,8 @@ public class CacheExtension implements Extension {
                             .scope(ApplicationScoped.class)
                             .addTransitiveTypeClosure(Cache.class)
                             .addTypes(types)
-                            .createWith(cc -> {
-                                return CacheBuilder.create(name, getConfigNode(qualifiers), keyType.rawType(),
-                                                           valueType.rawType());
-                            })
+                            .createWith(cc -> CacheBuilder.create(name, getConfigNode(qualifiers), keyType.rawType(),
+                                                                  valueType.rawType()))
                             .destroyWith((cache, context) -> cache.close());
                 }
             }

--- a/integrations/microstream/cdi/src/main/java/io/helidon/integrations/microstream/cdi/EmbeddedStorageManagerExtension.java
+++ b/integrations/microstream/cdi/src/main/java/io/helidon/integrations/microstream/cdi/EmbeddedStorageManagerExtension.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021 Oracle and/or its affiliates.
+ * Copyright (c) 2021, 2024 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -89,9 +89,7 @@ public class EmbeddedStorageManagerExtension implements Extension {
                                 .addTransitiveTypeClosure(EmbeddedStorageManager.class)
                                 .beanClass(EmbeddedStorageManager.class)
                                 .qualifiers(qualifiers)
-                                .createWith(cc -> {
-                                    return EmbeddedStorageManagerBuilder.create(getConfigNode(qualifiers)).start();
-                                })
+                                .createWith(cc -> EmbeddedStorageManagerBuilder.create(getConfigNode(qualifiers)).start())
                                 .destroyWith((storageManager, context) -> storageManager.shutdown());
                     }
                 }

--- a/integrations/microstream/cdi/src/main/java/module-info.java
+++ b/integrations/microstream/cdi/src/main/java/module-info.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021 Oracle and/or its affiliates.
+ * Copyright (c) 2021, 2024 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -29,10 +29,7 @@ module io.helidon.integrations.microstream.cdi {
     requires transitive jakarta.inject;
     requires jakarta.interceptor.api;
     requires jakarta.annotation;
-    requires microstream.base;
     requires microstream.cache;
-    requires microstream.persistence;
-    requires microstream.storage;
     requires microstream.storage.embedded;
 
     provides jakarta.enterprise.inject.spi.Extension

--- a/integrations/microstream/core/pom.xml
+++ b/integrations/microstream/core/pom.xml
@@ -1,6 +1,6 @@
 <!--
 
-    Copyright (c) 2021, 2023 Oracle and/or its affiliates.
+    Copyright (c) 2021, 2024 Oracle and/or its affiliates.
 
     Licensed under the Apache License, Version 2.0 (the "License");
     you may not use this file except in compliance with the License.
@@ -43,6 +43,10 @@
         <dependency>
             <groupId>one.microstream</groupId>
             <artifactId>microstream-configuration</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>one.microstream</groupId>
+            <artifactId>microstream-persistence-binary-jdk17</artifactId>
         </dependency>
         <dependency>
             <groupId>io.helidon.config</groupId>

--- a/integrations/microstream/core/src/main/java/module-info.java
+++ b/integrations/microstream/core/src/main/java/module-info.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021 Oracle and/or its affiliates.
+ * Copyright (c) 2021, 2024 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -28,5 +28,6 @@ module io.helidon.integrations.microstream {
     requires transitive microstream.storage;
     requires transitive microstream.storage.embedded;
     requires transitive microstream.storage.embedded.configuration;
+    requires microstream.persistence.binary.jdk17;
 
 }

--- a/integrations/microstream/metrics/src/main/java/io/helidon/integrations/microstream/metrics/MicrostreamMetricsSupport.java
+++ b/integrations/microstream/metrics/src/main/java/io/helidon/integrations/microstream/metrics/MicrostreamMetricsSupport.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021, 2022 Oracle and/or its affiliates.
+ * Copyright (c) 2021, 2024 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -39,7 +39,6 @@ import org.eclipse.microprofile.metrics.Tag;
 public class MicrostreamMetricsSupport {
 
     private static final String CONFIG_METRIC_ENABLED_VENDOR = "vendor.";
-    static final String BASE_ENABLED_KEY = CONFIG_METRIC_ENABLED_VENDOR + "enabled";
 
     private static final Metadata GLOBAL_FILE_COUNT = Metadata.builder()
             .withName("microstream.globalFileCount")

